dask/dataframe/core.py

```diff
@@ -714,6 +714,43 @@ def clear_divisions(self):
         divisions = (None,) * (self.npartitions + 1)
         return type(self)(self.dask, self._name, self._meta, divisions)
 
+    def enforce_runtime_divisions(self):
+        """Enforce the current divisions at runtime"""
+        if not self.known_divisions:
+            raise ValueError("No known divisions to enforce!")
+
+        def _check_divisions(df, expect):
```

Comment:
can we rather use 4 arguments instead of immediately unpacking?

Reply:
I'm pretty sure we would need to pass 4 distinct `BlockwiseDepDict` objects to `map_partitions` if we want to split the `expect` argument into 4 distinct args (which seems less readable to me). It is very possible that I am overthinking this, so let me know if you think I am missing something.

Reply:
If you find  the code clearer, we can do something like this:

```python
        def _check_divisions(df, id, expect_min, expect_max, last):
            # Check divisions
            real_min = df.index.min()
            real_max = df.index.max()
            # Upper division of the last partition is often set to
            # the max value. For all other partitions, the upper
            # division should be greater than the maximum value.
            valid_min = real_min >= expect_min
            valid_max = (real_max <= expect_max) if last else (real_max < expect_max)
            if not (valid_min and valid_max):
                raise RuntimeError(
                    f"`enforce_runtime_divisions` failed for partition {id}."
                    f" Expected a range of [{expect_min}, {expect_max}), "
                    f" but the real range was [{real_min}, {real_max}]."
                )
            return df

        ids, expect_mins, expect_maxes, islast = {}, {}, {}, {}
        for i, (dmin, dmax) in enumerate(
            zip(self.divisions[:-1], self.divisions[1:])
        ):
            key = (i,)
            ids[key] = i
            expect_mins[key] = dmin
            expect_maxes[key] = dmax
            islast[key] = i == (self.npartitions - 1)                      

        return self.map_partitions(
            _check_divisions,
            BlockwiseDepDict(ids),
            BlockwiseDepDict(expect_mins),
            BlockwiseDepDict(expect_maxes),
            BlockwiseDepDict(islast),
            meta=self._meta,
            enforce_metadata=False,
        )
```

Reply:
Ok I missed that, your initial solution seems clearer to me, so let's go with that